lib/matplotlib/backends/backend_svg.py

```diff
@@ -590,7 +595,7 @@ def _get_clip_attrs(self, gc):
         clippath, clippath_trans = gc.get_clip_path()
         if clippath is not None:
             clippath_trans = self._make_flip_transform(clippath_trans)
-            dictkey = (id(clippath), str(clippath_trans))
+            dictkey = (self._get_next_clip_id(clippath), str(clippath_trans))
         elif cliprect is not None:
```

Comment:
Another question, but perhaps outside of the scope of the changes for this PR: given that `cliprect` could be a more-performant code path, and maybe a more frequent one, shouldn't this condition be considered _before_ the `if clippath is None:` (_path_, not _rect_) conditional?

Reply:
Maybe, but it is very unclear to me that the micro-optimization actually matter.

With this testing order if an artist ends up with _both_ a clip path and a clip rect we respect the path instead (it is possible we already do validation of that someplace else so this is not a worry) of the rect.  Only changing the order without making sure that sort of validation existed could change user output.